src/arrayMethodJoin.js

```diff
@@ -4,8 +4,36 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(initialSeparator) {
+    let joinedArrString = '';
+    let separator = initialSeparator;
+    const newArr = [...this];
```

Comment:
Avoid creating a new array using the spread operator. You can directly use the original array.
But avoid array mutation